relay-profiling/src/lib.rs

```diff
@@ -127,17 +126,29 @@ struct MinimalProfile {
     version: sample::Version,
 }
 
-fn minimal_profile_from_json(payload: &[u8]) -> Result<MinimalProfile, ProfileError> {
+fn minimal_profile_from_json(
+    payload: &[u8],
+) -> Result<MinimalProfile, serde_path_to_error::Error<serde_json::Error>> {
     let d = &mut Deserializer::from_slice(payload);
-    serde_path_to_error::deserialize(d).map_err(ProfileError::InvalidJson)
+    serde_path_to_error::deserialize(d)
 }
 
-pub fn parse_metadata(payload: &[u8]) -> Result<(), ProfileError> {
+pub fn parse_metadata(payload: &[u8], event: Option<&Event>) -> Result<(), ProfileError> {
+    let (sdk_name, sdk_version) = match event {
+        Some(event) => (event.sdk_name(), event.sdk_version()),
+        _ => ("unknown", "unknown"),
+    };
     let profile = match minimal_profile_from_json(payload) {
         Ok(profile) => profile,
         Err(err) => {
-            relay_log::warn!(error = &err as &dyn Error, "invalid profile (minimal)");
-            return Err(err);
+            relay_log::warn!(
+                error = &err as &dyn Error,
+                "invalid profile (minimal, sdk: {} v{}). original: {}.",
+                sdk_name,
+                sdk_version,
+                err.inner(),
+            );
```

Comment:
Moving the SDK name and version to the message's tags means Sentry will group these errors under the same issue, and provide the ability to have insights into how many events are coming from what SDKs. Your call depending what you prefer hehe

Reply:
These messages will get to s4s in the form of warning messages. These are the same as regular messages captured with SDK's `capture_message` but with a different level -- `level:warning` in the issues page should allow you to filter these, for example. Just to be clear, no strong opinions here, just providing a different way you may like more.

Reply:
Ah, I didn't know I could do that. Perfect then.